herbert_core/classes/instrument_classes/instrument/detectors/@IX_detector_array/IX_detector_array.m

```diff
@@ -74,205 +128,412 @@
+                    tmp(i).x2 = val(nbeg(i):nend(i));
+                end
+            else
+                for i=1:numel(tmp)
+                    tmp(i).x2 = val;
+                end
+            end
+            obj.det_bank_ = tmp;
+            
+            if obj.do_check_combo_arg_
+                obj = obj.check_combo_arg();
+            end
+        end
+        
+        %---------------------------
+        function obj = set.phi (obj, val)
+            ndets = arrayfun (@(O)(O.ndet), obj.det_bank_);
+            % Input must be scalar or an array with ndet elements
+            if ~isnumeric(val) || ~(numel(val)==sum(ndets) || isscalar(val))
+                error('HERBERT:IX_detector_array:invalid_argument',...
+                    ['Scattering angle(s) must be a scalar or a ',...
+                    'numeric array with ', num2str(sum(ndets)), ' elements'])
+            end
+            % Alter every bank
+            tmp = obj.det_bank_;
+            if numel(val)>1
+                nend = cumsum(ndets);
+                nbeg = nend - ndets + 1;
+                for i=1:numel(tmp)
+                    tmp(i).phi = val(nbeg(i):nend(i));
+                end
+            else
+                for i=1:numel(tmp)
+                    tmp(i).phi = val;
+                end
+            end
+            obj.det_bank_ = tmp;
+            
+            if obj.do_check_combo_arg_
+                obj = obj.check_combo_arg();
+            end
         end
-
+        
+        %---------------------------
+        function obj = set.azim (obj, val)
+            ndets = arrayfun (@(O)(O.ndet), obj.det_bank_);
+            % Input must be scalar or an array with ndet elements
+            if ~isnumeric(val) || ~(numel(val)==sum(ndets) || isscalar(val))
+                error('HERBERT:IX_detector_array:invalid_argument',...
+                    ['Azimuthal angle(s) must be a scalar or a ',...
+                    'numeric array with ', num2str(sum(ndets)), ' elements'])
+            end
+            % Alter every bank
+            tmp = obj.det_bank_;
+            if numel(val)>1
+                nend = cumsum(ndets);
+                nbeg = nend - ndets + 1;
+                for i=1:numel(tmp)
+                    tmp(i).azim = val(nbeg(i):nend(i));
+                end
+            else
+                for i=1:numel(tmp)
+                    tmp(i).azim = val;
+                end
+            end
+            obj.det_bank_ = tmp;
+            
+            if obj.do_check_combo_arg_
+                obj = obj.check_combo_arg();
+            end
+        end
+        
+        %---------------------------
+        function obj = set.dmat (obj, val)
+            ndets = arrayfun (@(O)(O.ndet), obj.det_bank_);
+            % Input must have size [3,3,ndet] or [3,3] (==[3,3,1])
+            if ~isnumeric(val) || numel(size(val))>3 || ...
+                    (size(val,1)~=3 || size(val,2)~=3) || all(size(val,3)~=[1,sum(ndets)])
```

Comment:
```suggestion
            total_dets = sum(ndets);
            if ~(isnumeric(val) && isequal(size(val), [3 3 total_dets]) || ...
                                  (total_dets == 1 && isequal(size(val), [3 3]))
```

Reply:
Done. Slightly different to suggestion, but following the same clarity principle.